### ISSUE TITLE
Finalize USPS Address Flow UX/UI

### ISSUE THREAD

**Issue by bridgethapner** [department-of-veterans-affairs/va.gov-team#17563]
The UX/UI for the USPS address checking flow is started in ticket #13570. Review that ticket for a description of the feature. The interaction design of this flow needs a little more thought. This ticket is to finish up the flow.

I think what needs help here most is clarifying to the user what all the action items are. Users should be able to do some combination of the following as part of this flow: 

- Save the address they entered
- Accept the USPS suggestion
- Go back to edit their address

Bridget will send Sketch file with work done to date via Slack.

## Invision Updated Links
- Update unit number: https://adhoc.invisionapp.com/d/main#/console/20546460/441187431/preview
- Add Unit number: https://adhoc.invisionapp.com/d/main#/console/20546460/441187432/preview
- Confirmed address: https://adhoc.invisionapp.com/d/main#/console/20546460/441187433/preview
- Alternative options: https://adhoc.invisionapp.com/d/main#/console/20546460/441187434/preview

**Comment by msmanak** [comment]
![Slice.png](https://images.zenhubusercontent.com/5faabf11b099434179a66d66/97dc8fe9-c915-4f3e-ac73-462e6eebc966)

For @bridgethapner review

**Comment by bridgethapner** [comment]
@msmanak Thank you for this! I like the use of the outlined style of the Large Button CCs rather than 'Edit address' as a link in text. When you finalize can you add to the Invision & link to where the screens are in the prototype here?

**Comment by msmanak** [comment]
Reopening this as @bessgreen bought up some interesting questions regarding to edit/accept state. @lexicalninja and I decided to revise "Edit Mailing Address (alternative options)" to offer both options to edit the address or overide what USPS suggested.

![Slice-Edit Mailing Address (alternative options).png](https://images.zenhubusercontent.com/5faabf11b099434179a66d66/e692aad3-6f71-45f7-82ee-3cab4b6b5cb5)

**Comment by bridgethapner** [comment]
I chatted with @bessgreen and we are all set here on the UI, final screens are in Invision. Closing this ticket.

**Comment by lexicalninja** [comment]
A few of the content blurbs say things about the USPS and we should probably remove them since we use a different system. We can probably just say we can't verify your address without mentioning who we are verifying with, but content folks should be able to clear that up for us

**Comment by lexicalninja** [comment]
I tested this last week for Meli to get an idea of what web does and va.gov doesn't actually do what the product guide indicates. Current staging and production on web just changes the address without user input and without telling the user that it is using a different address than is entered

**Comment by lexicalninja** [comment]
I didn't see any of the validation and I tried to break it. When I put in a bad city, it just picked a city

But depending on what input I had for city, it would pick different cities as the answer. It appears it just takes the highest ranked address from the service and uses that.

The reason Meli built this out is that she wanted to improve the UX from what is currently on the web, but harness the product guide and then figure out what the validation api returned. 

the api returns any validation errors for the fields and if they pass, either says the address is good or returns a list of addresses that are ranked by a "confidence score". There is no indication of how many can actually come back, so Meli thought that anything more that 2 or 3 was too much. 

Basically, web wasn't doing what the guide said it was and she felt that this was one of the opportunities that Steve mentioned that we could go above and beyond web and this is what she came up with. When we looked at the api and web and the guide this felt like the middle ground of the tech and content we had that gave the best Veteran experience.

**Comment by bridgethapner** [comment]
Alright, I'm not up to date here and I don't understand what the experience on VA.gov is (if it doesn't match what's in the product guide) and what we need to to to change it. We need to take a step back and understand what the current feature is (or what it's supposed to be if something is broken) on VA.gov so we can understand what we're working from & what we want our product feature to be. 

Either way - @bessgreen cc'ing you here to let you know the desired feature for address validation is likely not finalized, we're still working some things out

**Comment by bridgethapner** [comment]
Circling back here -- 

- Per [this thread](https://dsva.slack.com/archives/C909ZG2BB/p1609860194334900) in DSVA, our app will use the same front-end logic as the Profile VA.gov team that oftentimes skips over address validation IF the user enters an address that's close enough. That's what we learned today that was the missing piece of understanding the VA.gov feature.
- The UX/UI for the address validation scenarios we've developed so far is done & doesn't need further edits.
